packages/rich-text/src/component/use-input-and-selection.js

```diff
@@ -140,6 +140,15 @@ export function useInputAndSelection( props ) {
 			// for the rich text instance that contains the start or end of the
 			// selection.
 			if ( ownerDocument.activeElement !== element ) {
+				// Only process if the active elment is contentEditable, either
+				// this rich text instance or the writing flow parent. Fixes a
+				// bug in Firefox where it strangely selects the closest
+				// contentEditable element, even though the click was outside
+				// any contentEditable element.
+				if ( ownerDocument.activeElement.contentEditable !== 'true' ) {
```

Comment:
Do we know if this popped recently as a bug in FF itself?

Reply:
No idea. Could have been there all along. This code was introduced relatively recently for partial multi selection.